### ISSUE TITLE
Can't "pip install ravenpy[gis]" on ARM-based Mac architectures

### ISSUE THREAD

**Issue by julemai** [CSHS-CWRA/RavenPy#370]
* RavenPy version: (whatever comes currently with pip install; ravenpy-0.14.1)
* Python version: 3.10.13
* Operating System: MacOS Sanoma 14.2.1 (23C71) on Apple M2 Max

### Description

I try to `pip install RavenPy[gis]` and it does throw error messages. I am guessing because there is and issue with compiling the Raven code, i.e.,

```
      /private/var/folders/mc/ks_z5qjs1bq7m6kgtlx1_v940000gn/T/pip-install-ygjkwybk/raven-hydro_b3e722675cf448a4a4531ae189487ef1/RavenHydroFramework/src/SubBasin.cpp:1784:25: error: use of undeclared identifier 'finite'; did you mean 'isfinite'?
            ExitGracefullyIf(!_finite(_aUnitHydro[n]),
                              ^
      /private/var/folders/mc/ks_z5qjs1bq7m6kgtlx1_v940000gn/T/pip-install-ygjkwybk/raven-hydro_b3e722675cf448a4a4531ae189487ef1/RavenHydroFramework/src/RavenInclude.h:1468:20: note: expanded from macro '_finite'
      #define _finite(v) finite(v)
                         ^
      /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.5.sdk/usr/include/c++/v1/math.h:403:80: note: 'isfinite' declared here
      _LIBCPP_NODISCARD_EXT _LIBCPP_CONSTEXPR_SINCE_CXX23 _LIBCPP_HIDE_FROM_ABI bool isfinite(_A1 __x) _NOEXCEPT {
                                                                                     ^
      1 warning and 1 error generated.
```

The full output is attached.

[compile_msg.txt](https://github.com/CSHS-CWRA/RavenPy/files/15500861/compile_msg.txt)

**Comment by Zeitsperre** [comment]
Hey Julie,

Are you using a new architecture Mac? The issue has to do with the newer standards for `clang` and is a problem with `RavenHydroFramework`. James just released released a new version that should let us add a few more build architectures (I'm out of the office at the moment, but will be doing that when I get back), but apparently the problem requires a bit more effort for the ARM-based Macs. He's aware of the problem.

**Comment by julemai** [comment]
If the user provides an executable, I'd hope it also runs. :D But that's probably not what you mean.

**Comment by huard** [comment]
One subtlety is that by controlling the build, we're able to create a shared library that let's us interact directly with the RavenC library. It's very superficial for now, but I'm hoping we can use this strategy and the BMI interface to reduce IO and improve performance in the future.

**Comment by julemai** [comment]
Hi everyone,

I just tried again to install `ravenpy` with pip on another Mac and got another set of error messages/warnings. Thought of sharing this in case that's of interest.

I tried:
```
pip install ravenpy[gis]
```

One error is that the shared HDF5 lib is not found. HDF5 is installed on my machine. I have no idea why it is not found. Happy to provide more information if needed. Attached is the full terminal output.

Cheers!

[error-ravenpy.txt](https://github.com/user-attachments/files/18495805/error-ravenpy.txt)
